### DISCUSSION TITLE
iOS archive fails because of missing iOS App development provisioning profile

### DISCUSSION THREAD

**Discussion by aureliencolas** [fastlane/fastlane#20390] [Q&A]
In order to archive and export iOS binaries with Fastlane, I'm using Match with appstore certificates and provisioning profiles stored on bitbucket and until then it worked fine.

However, for some reason, I've decided to clear all the provisioning profiles on my computer (in ~/Library/MobileDevice/Provisioning\ Profiles), thinking match would retrieve the needed ones when archiving the apps with Fastlane.

From now on, although match properly retrieves the certificate and provisioning profile for the app being built, xcodebuild fails because Xcode does not find any **development** provisioning profile for this app. The only way I've found to solve the issue is to launch Xcode and let it create the development profile (Automatic signing is enabled) and then build the app with Fastlane again.

Here are some of the build logs:
`[21:06:17]: Detected provisioning profile mapping: {:"com.gibtele.iptv"=>"match AppStore com.gibtele.iptv"}`
`...`
`[21:06:42]: ▸ ❌  error: No profiles for 'com.gibtele.iptv' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.gibtele.iptv'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'Generic_iOS' from project 'videofutur')`

I've tried to sync **development** cert and profile with Match too but it does not help.
Clearly I've missed something in the configuration of Fastlane/Match. Could someone help me on this? Is it normal that the develoment provisioning profile is required to archive an iOS app?

**Comment by harrimaatta** [comment]
Did you @aureliencolas ever found solution for this? What was the key thing to change/update/...?

**Reply by aureliencolas** [reply]
Well I found a way to bypass this error but not to really fix it.

My problem was that dev profiles were not necessarily up to date on my mac when launching the build lane and I had to launch xcode first, which automatically downloads up-to-date profiles, and only then launch the build lane.

I fixed my issue using the `xcargs` parameter of `build_app()` in order to allow xcodebuild to update provisioning profiles. Here is the exact syntax:
`build_app(scheme: scheme, output_directory: output_directory, output_name: output_name, clean: true, silent: false, xcargs: "-allowProvisioningUpdates")`

Hope it helps.
